cron-backup/nodered_backup.sh

```diff
@@ -4,55 +4,55 @@
 	exit
 else
-	tar cvzf /var/lib/backup/"${SOURCE_NAME}"_nodered_data_backup_"${DATE1}".tgz ${nodered_src}/
+	tar cvzf /var/lib/backup/nodered/"${SOURCE_NAME}"_nodered_data_backup_"${DATE1}".tgz ${nodered_src}/
 fi
 
-s3cmd sync --no-mime-magic /var/lib/backup/*nodered*.tgz s3://"${S3_BUCKET_NODERED}"/
+s3cmd put -r --no-mime-magic /var/lib/backup/nodered/ s3://"${S3_BUCKET_NODERED}"/
 
 # Moving the backup to S3 bucket
 if [ $? -eq 0 ]; then
 
-        echo "DATE:" $DATE > /node_graf_nginx.txt
-        echo " " >> /node_graf_nginx.txt
-        echo "DESCRIPTION: ${SOURCE_NAME}_Nodered backup" >> /node_graf_nginx.txt
-        echo " " >> /node_graf_nginx.txt
-        echo "STATUS: Node-red backup is Successful." >> /node_graf_nginx.txt
-        echo " " >> /node_graf_nginx.txt
-        echo "******* Node-red Data Backup ****************" >> /node_graf_nginx.txt
-        echo " " >> /node_graf_nginx.txt
-        s3cmd ls --no-mime-magic s3://"${S3_BUCKET_NODERED}"/  --human-readable | grep -i "${SOURCE_NAME}"_nodered_data | cut -d' ' -f3- | tac | head -10 | sed "s,s3:\/\/${S3_BUCKET_NODERED}\/,,g" &>> /node_graf_nginx.txt
-        echo " " >> /node_graf_nginx.txt
-        echo "************** END **************************" >> /node_graf_nginx.txt
+        echo "DATE:" $DATE > /tmp/nodered.txt
+        echo " " >> /tmp/nodered.txt
+        echo "DESCRIPTION: ${SOURCE_NAME}_Nodered backup" >> /tmp/nodered.txt
+        echo " " >> /tmp/nodered.txt
+        echo "STATUS: Node-red backup is Successful." >> /tmp/nodered.txt
+        echo " " >> /tmp/nodered.txt
+        echo "******* Node-red Data Backup ****************" >> /tmp/nodered.txt
+        echo " " >> /tmp/nodered.txt
+        s3cmd ls --no-mime-magic s3://"${S3_BUCKET_NODERED}"/  --human-readable | grep -i "${SOURCE_NAME}"_nodered_data | cut -d' ' -f3- | tac | head -10 | sed "s,s3:\/\/${S3_BUCKET_NODERED}\/,,g" &>> /tmp/nodered.txt
+        echo " " >> /tmp/nodered.txt
+        echo "************** END **************************" >> /tmp/nodered.txt
 
 else
-        echo "DATE:" "$DATE" > /node_graf_nginx.txt
-        echo " " >> /node_graf_nginx.txt
-        echo "DESCRIPTION: ${SOURCE_NAME}_Nodered backup" >> /node_graf_nginx.txt
-        echo " " >> /node_graf_nginx.txt
-        echo "STATUS: Nodered backup is Failed." >> /node_graf_nginx.txt
-        echo " " >> /node_graf_nginx.txt
-        echo "Something went wrong, Please check it"  >> /node_graf_nginx.txt
-        < /node_graf_nginx.txt mail -s "${SOURCE_NAME}: Nodered Data Backup" "${CRON_BACKUP_MAIL}"
+        echo "DATE:" "$DATE" > /tmp/nodered.txt
+        echo " " >> /tmp/nodered.txt
+        echo "DESCRIPTION: ${SOURCE_NAME}_Nodered backup" >> /tmp/nodered.txt
+        echo " " >> /tmp/nodered.txt
+        echo "STATUS: Nodered backup is Failed." >> /tmp/nodered.txt
+        echo " " >> /tmp/nodered.txt
+        echo "Something went wrong, Please check it"  >> /tmp/nodered.txt
```

Comment:
Too much repetition of identical content. "Don't repeat yourself." Consider refactoring so that you don't have all these messages thousands of times, so it's easier to enforce consistency and apply stylistic changes.

Reply:
ok.